src/compiler/checker.ts

```diff
@@ -10713,13 +10713,18 @@ namespace ts {
             checkGrammarJsxElement(node);
             checkJsxPreconditions(node);
 
-            // The reactNamespace symbol should be marked as 'used' so we don't incorrectly elide its import. And if there
-            // is no reactNamespace symbol in scope when targeting React emit, we should issue an error.
-            const reactRefErr = compilerOptions.jsx === JsxEmit.React ? Diagnostics.Cannot_find_name_0 : undefined;
-            const reactNamespace = compilerOptions.reactNamespace ? compilerOptions.reactNamespace : "React";
-            const reactSym = resolveName(node.tagName, reactNamespace, SymbolFlags.Value, reactRefErr, reactNamespace);
-            if (reactSym) {
-                getSymbolLinks(reactSym).referenced = true;
+            // The JSX factory namespace symbol should be marked as 'used' so we don't incorrectly elide its import. And if
+            // it isn't in scope when targeting React emit, we should issue an error.
+            const jsxFactoryRefErr = compilerOptions.jsx === JsxEmit.React ? Diagnostics.Cannot_find_name_0 : undefined;
+            let jsxFactoryNamespace = "React";
+            if (compilerOptions.jsxFactory) {
+                jsxFactoryNamespace = compilerOptions.jsxFactory.split(".")[0];

```

Comment:
I was using treating `jsxFactory` as an `Identifier` in `factory.ts` as a convenience in the `a.b.c` case, but really in that scenario it should be a `PropertyAccessExpression`.

In that case I only want `a` out of `a.b.c` so that I can set `referenced = true` on that symbol alone.